lib/ex_unit/test/ex_unit/capture_io_test.exs

```diff
@@ -28,6 +28,22 @@ defmodule ExUnit.CaptureIOTest do
   import ExUnit.CaptureIO
   doctest ExUnit.CaptureIO, import: true
 
+  test "no leakage on failures" do
+    processes_count = length(Process.list())
+    group_leader = Process.group_leader()
+
+    for _ <- 1..10 do
+      assert_raise ArgumentError, fn ->
+        capture_io(fn ->
+          raise ArgumentError
```

Comment:
Using `Process.list()` is not an ideal because something else could be started. To check that the `StringIO` process is closed, could fetch the `group_leader` pid (`StringIO`) inside the `capture_io` fun and check the pid has closed after the `assert_raise` using a monitor, sending the reference to self , receiving the reference and `assert_receive` the `:DOWN` message.